sips/adapt_fees_to_declared_effective_balance.md

```diff
@@ -0,0 +1,166 @@
+
+## Motivation
+
+Operators on SSV Network charge validators a fee based on an expected workload. The expected workload is proportional to the the effective balance a validator has. Currently the effective balance is assumed to be a constant 32 ETH by the contract. With the introduction of [EIP-7251](https://eips.ethereum.org/EIPS/eip-7251) the effective balance of a validator could be as high as 2048 ETH.
+
+The same reasoning applies for the network fee charged to each validator registered.
+
+
+## Rationale
+
+Do not perform duties if the validator's effective balance is greater by more than a deviation configured by the contract's owner. This will incentivize validators to always register the correct amount on the contract. Effective balance may grow due to compounding rewards and it is the user's responsibility to update their effective balance before their operators pause. Paused operators will resume when the effective balance is updated on the contract.
+
+## Specification
+
+### Contract
+When registering new validators, add a new parameter to indicate the current effective balance of the validator being added. This will be used to update operators' [snapshots](https://github.com/ssvlabs/ssv-network/blob/583b7b7cb1c1abc5d4c3b13bafca59bf315113b6/contracts/interfaces/ISSVNetworkCore.sol#L10) with the right amounts, calculate the network fee applied to the validator, and emit the event with this information to allow the node confirm it.
+
+
+#### Adapted functions
+
+```solidity
+function registerValidator(
+    bytes calldata publicKey,
+    uint64[] memory operatorIds,
+    bytes calldata sharesData,
+    uint256 ethEffectiveBalance, \\ This is new
+    uint256 amount,
+    Cluster memory cluster
+) external
+```
+```solidity
+function bulkRegisterValidator(
+    bytes[] memory publicKeys,
+    uint64[] memory operatorIds,
+    bytes[] calldata sharesData,
+    uint256[] ethEffectiveBalance, \\ This is new
+    uint256 amount,
+    Cluster memory cluster
+) external
+```
+
+
+    
+#### Adapted event
+```solidity
+event ValidatorAdded(
+    address indexed owner,
+    uint64[] operatorIds,
+    bytes publicKey,
+    bytes shares,
+    uint256 ethEffectiveBalance, //This is new
+    Cluster cluster);
+```
+
+### New functions
+
+```solidity
+// Function to update the effective balance for a set of validators
+function updateEthEffectiveBalance(
+    bytes[] memory publicKeys,
+    uint64[] memory operatorIds,
+    uint256[] newEffectiveBalance,
+    Cluster memory cluster
+) external
+```
+```solidity
+// Function to update the maximum deviation allowed for the ETH 
+// balance declared by validator owners
+// @dev Using 10000 to represent 2 digit precision: 100% = 10000, 10% = 1000
+// @dev Only the owner of the contract can perform the update
+function updateAllowedBalanceDeviation(uint64 percentage) external onlyOwner
+```
+
+```solidity
+// Function to get the maximum ETH balance deviation
```

Comment:
does it really need to be stored on chain? or can nodes simply parse it from events?

Reply:
events sounds good to me

@mtabasco wants to comment?

Reply:
Yeah, if there is no major UX issue removing `getAllowedBalanceDeviation()`, the ETH balance deviation can be emitted.